web/core/store/favorite.store.ts

```diff
@@ -0,0 +1,233 @@
+  removeFavoriteEntity: (workspaceSlug: string, entityId: string) => Promise<void>;
+}
+
+export class FavoriteStore implements IFavoriteStore {
+  // observables
+  favoriteIds: string[] = [];
+  favoriteMap: {
+    [favoriteId: string]: IFavorite;
+  } = {};
+  entityMap: {
+    [entityId: string]: IFavorite;
+  } = {};
+  // service
+  favoriteService;
+
+  constructor() {
+    makeObservable(this, {
+      // observable
+      favoriteMap: observable,
+      entityMap: observable,
+      favoriteIds: observable,
+      // action
+      fetchFavorite: action,
+      // CRUD actions
+      addFavorite: action,
+      getGroupedFavorites: action,
+      moveFavorite: action,
+      removeFavoriteEntity: action,
+    });
+    this.favoriteService = new FavoriteService();
+  }
+
+  /**
+   * Creates a favorite in the workspace and adds it to the store
+   * @param workspaceSlug
+   * @param data
+   * @returns Promise<IFavorite>
+   */
+  addFavorite = async (workspaceSlug: string, data: Partial<IFavorite>) => {
+    try {
+      data = { ...data, parent: null, is_folder: data.entity_type === "folder" };
+      const response = await this.favoriteService.addFavorite(workspaceSlug, data);
+      runInAction(() => {
+        set(this.favoriteMap, [response.id], response);
+        response.entity_identifier && set(this.entityMap, [response.entity_identifier], response);
+        this.favoriteIds = [response.id, ...this.favoriteIds];
+      });
+      return response;
+    } catch (error) {
+      console.log("Failed to create favorite from favorite store");
+      throw error;
+    }
+  };
+
+  /**
+   * Updates a favorite in the workspace and updates the store
+   * @param workspaceSlug
+   * @param favoriteId
+   * @param data
+   * @returns Promise<IFavorite>
+   */
+  updateFavorite = async (workspaceSlug: string, favoriteId: string, data: Partial<IFavorite>) => {
+    try {
+      const response = await this.favoriteService.updateFavorite(workspaceSlug, favoriteId, data);
+      runInAction(() => {
+        set(this.favoriteMap, [response.id], response);
+      });
+      return response;
+    } catch (error) {
+      console.log("Failed to update favorite from favorite store");
+      throw error;
+    }
+  };
+
+  /**
+   * Moves a favorite in the workspace and updates the store
+   * @param workspaceSlug
+   * @param favoriteId
+   * @param data
+   * @returns Promise<void>
+   */
+  moveFavorite = async (workspaceSlug: string, favoriteId: string, data: Partial<IFavorite>) => {
+    try {
+      const response = await this.favoriteService.updateFavorite(workspaceSlug, favoriteId, data);
+      runInAction(() => {
+        // add the favorite to the new parent
+        if (!data.parent) return;
+        set(this.favoriteMap, [data.parent], {
+          ...this.favoriteMap[data.parent],
+          children: [response, ...this.favoriteMap[data.parent].children],
+        });
```

Comment:
can we set children directly instead of changing reference of the favorite object.
```suggestion
        set(this.favoriteMap, [data.parent, "children"], [response, ...this.favoriteMap[data.parent].children]);
```

please check for the similar ones below on this file